packages/@sanity/client/src/datasets/datasetsClient.js

```diff
@@ -22,6 +22,12 @@ assign(DatasetsClient.prototype, {
     return this.request({uri: '/datasets'})
   },
 
+  copy(name, targetDataset) {
+    validate.dataset(name)
+    validate.dataset(targetDataset)
+    return this.request({method: 'PUT', uri: `/datasets/${name}/copy`, body: {targetDataset}})
+  },
+
```

Comment:
I'm actually prefer if we could just call `client.request()` in the CLI for this. If we do expose a client method, I think it should listen for progress events and either emit those as an observable, or simply wait for completed task/error and return a promise that resolves/rejects.

Reply:
Ah, what you mean is that we don't create a dedicated `copy` command for this?

Reply:
I doubt it would be preferable to clone the `validate` code, as that lives in the `client` package, and I would need this to be done in `core`. Do we have a smart way to use this code from the client, or do I have to copy the functionality (although it isn't very big).

Reply:
> Ah, what you mean is that we don't create a dedicated `copy` command for this?

Yep.

Reply:
> I doubt it would be preferable to clone the `validate` code, as that lives in the `client` package, and I would need this to be done in `core`. Do we have a smart way to use this code from the client, or do I have to copy the functionality (although it isn't very big).

You could either copy it or just leave it up to the API to respond with an error on invalid dataset names - I assume it's doing validation on the server side anyway? :)

Reply:
Yes, of course. I'll think about duplicating it and adding it to the client side.